### PR TITLE
Add supporting scripts for running automation in Azure Batch

### PR DESCRIPTION
**Pull request by troydai**
## Add supporting scripts for automation 
1. Build scripts for build both product and test codes.
2. Application scripts for dispatch and set up tests in Azure Batch.

## Update product test codes 
1. Make tests themselves packages.
2. Remove docstring in test methods so that when listing tests using
nosetests, their method names are exposed correctly.

### CODE REVIEW

File: src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_shell_progress.py [hunk 1]
```diff
@@ -24,6 +25,7 @@ def test_shell_progress_write(self):
         self.assertEqual(prog.get_progress_message(), 'Test1')
         self.assertEqual(prog.PROGRESS_BAR, "")
 
+    @unittest.skipIf(os.getenv('TERM') is None, 'Skip when $TERM is missing')
```
Comment (troydai) [inline src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_shell_progress.py]: @oakeyc @derekbekoe when this test is run in the container, the `tput` command will fail. You have any workaround?

File: scripts/batch/build_all.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    'Programming Language :: Python :: 3.4',
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
+    'License :: OSI Approved :: MIT License',
+]
+
+DEPENDENCIES = [
+    'azure-cli',
+    'azure-cli-testsdk',
+    'azure-devtools'
```
Comment (derekbekoe) [inline scripts/batch/build_all.sh]: pin versions?

File: scripts/batch/build_all.sh [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    description='Microsoft Azure Command-Line Tools Full Tests',
+    license='MIT',
+    author='Microsoft Corporation',
+    author_email='azpycli@microsoft.com',
+    url='https://github.com/Azure/azure-cli',
+    zip_safe=False,
+    classifiers=CLASSIFIERS,
+    packages=[
+        'azure.cli.core.tests',
+EOL
```
Comment (derekbekoe) [inline scripts/batch/build_all.sh]: Looks like this is cut short?

File: scripts/batch/build_all.sh [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                       '**/*.cer',
+                       '**/*.pem', 
+                       '**/*.pfx',
+                       '**/*.txt',
+                       '**/*.json',
+                       '**/*.byok',
+                       '**/*.js',
+                       '**/*.md',
+                       '**/*.bat',
+                       '**/*.txt']},
```
Comment (derekbekoe) [inline scripts/batch/build_all.sh]: What's this for?

File: src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_shell_progress.py (hunk 1 above)
Comment (derekbekoe) [inline src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_shell_progress.py]: `skipUnless(os.getenv('TERM'))`??

File: scripts/batch/build_all.sh (hunk 2 above)
Comment (troydai) [inline scripts/batch/build_all.sh]: actually, should be removed. `azure-cli-testsdk` will pull it in.

File: scripts/batch/build_all.sh (hunk 3 above)
Comment (troydai) [inline scripts/batch/build_all.sh]: no. following lines are a loop adding all the test packages to the setup.py.

File: scripts/batch/build_all.sh (hunk 4 above)
Comment (troydai) [inline scripts/batch/build_all.sh]: these are test data file.

File: src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_shell_progress.py (hunk 1 above)
Comment (troydai) [inline src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_shell_progress.py]: not sure if empty string is legal or not.

**Pull request merged**